### PR TITLE
feat: initial api explorer dev portal support

### PR DESCRIPTION
**Pull request by bryans99**
# Description

Allows API explorer to run without a banner inside of the developer portal. Note that this will eventually be refactored to build a version of API explorer inside of the developer portal. For now API explorer is built in sdk codegen and the resultant javascript build is copied to the developer portal.

# Changes

1. Hard coded check for dev portal url to set headless parameter and base url for router. 
2. Headless parameter added to API explorer. When running in headless mode, selectors are moved to the sidebar. Current header is not shown.
3. Selectors have been wrapped to allow simplified use in two places (header or sidebar)
4. Tests updated

## To dos

1. Specific tests for headless
2. Externalize sidebar toggle such that dev portal can open/close API explorer sidebar (diff closes sidebar so a page reload is required to redisplay sidebar)
3. Fix diff navigation - occasionally results in an error or does not work.

## Running in dev portal

image

## Normal standalone

image

### CODE REVIEW

File: packages/api-explorer/src/reducers/spec/utils.ts [hunk 1]
```diff
@@ -126,7 +126,12 @@ export const fetchSpec = (key: string, specs: SpecList): SpecState => {
  * @param specs to use to find the default spec key
  */
 export const getSpecKey = (location: AbstractLocation, specs?: SpecList) => {
-  const pathNodes = location.pathname.split('/')
+  const pathName = location.pathname
+  // if (pathName.startsWith('/api/docs')) {
+  //     pathName = pathName.replace('/api/docs', '')
+  // }
+  // console.log({locationPathname: location.pathname, pathName, specs})
```
Comment (jkaster) [inline packages/api-explorer/src/reducers/spec/utils.ts:133]: Cruft to clean up next time?

**Review by jkaster — APPROVED** [review]
LGTM. Thanks for the detailed PR description.

**Review by jkaster — APPROVED** [review]

**Review by noahamac — APPROVED** [review]
Exemplary PR... Nice work

**Review by jkaster — APPROVED** [review]
Looks good. You made a big dent!

**Pull request merged**